src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -69,16 +69,17 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			experimentalUseConpty: useConpty
 		};
 
-		// TODO: Need to verify whether executable is on $PATH, otherwise things like cmd.exe will break
-		// fs.stat(shellLaunchConfig.executable!, (err) => {
-		// 	if (err && err.code === 'ENOENT') {
-		// 		this._exitCode = SHELL_PATH_INVALID_EXIT_CODE;
-		// 		this._queueProcessExit();
-		// 		this._processStartupComplete = Promise.resolve(undefined);
-		// 		return;
-		// 	}
-		this.setupPtyProcess(shellLaunchConfig, options);
-		// });
+		let envPathVariable: string | undefined = process.env.PATH;
+
+		fs.stat(shellLaunchConfig.executable!, (err) => {
+			if (err && err.code === 'ENOENT' && envPathVariable!.search(shellLaunchConfig.executable!) === -1) {
```

Comment:
This won't work because executable is a file and $PATH just contains a bunch of folders, it would need to check the contents of every folder inside $PATH for executable, iff it's not an absolute path.

Reply:
Done.